### ISSUE TITLE
Trouble opening git-managed workspace

### ISSUE THREAD

**Issue by stephankoe** [Zettlr/Zettlr#4540]
### Description

I'm using Zettlr on Ubuntu 23.04 and git for version management. It generally worked fine, until I was switching branches in git. After the folder view didn't automatically load the new files, I re-opened Zettlr, but the window remained empty and the error message `system.error.open_root_error` appeared in the system's notifications. All menu options (file, edit, ...) initially only showed the "Quit" and "Inspect" options. Later, I was able to open the log viewer where I found the following messages:

```
Warning: [Assets Provider] Found outdated filter links.lua; copying...
Warning: [Assets Provider] Found outdated filter links.lua; copying...
Error: Could not open new root /home/user/Documents/MyWorkspace
```

Just now, after having waited for about 10 minutes with the empty Zettlr window open, the workspace suddenly opened. The log viewer shows more messages now:

```
Warning: [Translation Provider] Could not update translations: ETIMEDOUT
Error: [Update Provider] ETIMEDOUT: Couldn't check for updates. RequestError: connect ETIMEDOUT 45.129.xxx.yyy:443
Error: [R] [Main Window] The descriptor for path /home/user/Documents/MyWorkspace/path/to/hello world was empty! (index.js:2)
```

I have to mention that the file `/home/user/Documents/MyWorkspace/path/to/hello world` contains a space and that the Internet connection here is not very reliable. However, when I restart Zettlr, opening the workspace takes a long time again.

### Reproducing

The error frequently occurs after switching branches in git.

### Zettlr Version

Stable (most recent version)

### Specify version

2.3.0

### Installation Method

Flathub

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Ubuntu 23.04

**Comment by nathanlesage** [comment]
The major reason you had to wait so long is that Zettlr has to parse all the (now changed) files. It detects that tons of files have changed, and so it takes a long time to parse everything anew. Zettlr can be a little wonky when it comes to changing a large amount of files quickly, so especially if your managed workspace is large, I do recommend switching branches (especially if there are large differences) while Zettlr is closed.

Additionally, I invite you to try out the current betas to check how the experience differs.

**Comment by stephankoe** [comment]
Ok. My current workspace contains 173 files, with the `.git` folder it's 450. My branches differ only in a handful of files. Does Zettlr parse all files once or every time I open Zettlr? Apparently, changing branches with Zettlr closed also exhibits this problem. I'll try the beta version to check whether it also exhibits this problem.

**Comment by nathanlesage** [comment]
Well, it depends a little on how git switches the files: if it manipulated the mod times correctly, Zettlr only reparses those with changed modtimes. But I'm not too deep into that architecture to answer this reliably.

**Comment by stephankoe** [comment]
I just tried the AppImage version of Zettlr 3.0.0-beta7 with default settings. It appears to work, even after changing branches (while Zettlr was closed). But so does the installed version now. I start to believe, it might not be related to git after all. Is there any location in the code that could block if there's network available, but the servers aren't reachable? I'm currently in China, and unpredictable delays and package loss aren't uncommon when accessing foreign servers through the so-called "Great Firewall". I'll keep an eye open the next days and update this issue when I spot some patterns.

Besides, when I switch branches while Zettlr is open, the files that weren't in the original branch appear twice in the file list. When I click on one of them, both copies get highlighted simultaneously.

**Comment by stephankoe** [comment]
This morning, the installed version of Zettlr 2.3.0 again takes a long time to open my workspace. But the AppImage version of Zettlr 3.0.0-beta7 opens it instantly with the open workspaces and all available menu options.

**Comment by stephankoe** [comment]
To rule out the possibility that the delayed opening is caused by my configuration, I downloaded and ran the AppImage of Zettlr 2.3.0 with it's default configuration. It exhibits the same behavior, i.e., it took several minutes before it opened the default tutorial workspace. Hence, I guess it's safe to assume that this issue was already resolved anywhere in 3.0.0, so I'll close this issue as resolved.
